new Leetcode/49. Group Anagrams/49. Group Anagrams.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Created by sindhuhari on 1/19/18.
+ */
+var groupAnagrams = function (strs) {
+    if(!strs || strs.length === 0) {
+        return [];
+    }
+    var mapping = {};
+    var oldArray = strs.slice();
+    var sortStrings = strs;
+    var result = [];
+    for(var i = 0; i < sortStrings.length; i++) {
+        sortStrings[i] = sortString(sortStrings[i]);
+        if(mapping[sortStrings[i]]) {
+            mapping[sortStrings[i]].push(i)
+        } else {
+            mapping[sortStrings[i]] = [];
+            mapping[sortStrings[i]].push(i)
+        }
+    }
+    console.log(mapping)
+    for(var key in mapping) {
+        var group = [];
+        for(var value = 0 ; value < mapping[key].length; value++) {
+            var index = oldArray[mapping[key][value]];
+            group.push(index)
+        }
+        console.log(group)
+        result.push(group)
+    }
+    return result
+};
+
+var sortString = function(text) {
+    return text.split('').sort().join('');
```

Comment:
can we do it better than this ? your algos complexity is O(k * nlogn) where n is the length of each string. 

try doing it in O(k * n). hint: maintain a count of each character you encounter. 
if the string is "extraaaahello"  -> {a: 4, e: 2, l: 2, t: 1, r: 1, o: 1, x: 1}. this can be done in O(n).